lib/lightning/projects.ex

```diff
@@ -697,4 +698,125 @@ defmodule Lightning.Projects do
   defp delete_history_for(_project) do
     {:error, :missing_history_retention_period}
   end
+
+  def invite_collaborators(project, collaborators, inviter) do
+    Multi.new()
+    |> Multi.put(:collaborators, collaborators)
+    |> Multi.merge(&register_users/1)
+    |> Multi.run(:add_users_to_project, fn _repo, changes ->
+      add_users_to_project(changes, project, collaborators)
```

Comment:
instead of propagating the whole changes, could we pass only the data that is needed to add the user to a project?